src/supervisor/mod.rs

```diff
@@ -198,135 +211,320 @@ async fn terminate_children<'a>(
     ev_notifier: &mut EventNotifier,
     runtime_children: impl Iterator<Item = Worker>,
 ) -> (Vec<worker::Spec>, Result<(), TerminationError>) {
-    let mut children_specs = Vec::new();
+    let mut spec_children = Vec::new();
     let mut termination_errors: HashMap<String, Arc<worker::TerminationError>> = HashMap::new();
     for runtime_child in runtime_children {
         let child_runtime_name = runtime_child.runtime_name.to_owned();
         match runtime_child.terminate().await {
             Err((failed_child_spec, termination_err)) => {
+                println!("(1) DEBUG -- worker failed");
                 let termination_err = Arc::new(termination_err);
                 ev_notifier
                     .worker_termination_failed(child_runtime_name.clone(), termination_err.clone())
                     .await;
                 termination_errors.insert(child_runtime_name, termination_err);
-                children_specs.push(failed_child_spec);
+                spec_children.push(failed_child_spec);
             }
             Ok(failed_child_spec) => {
+                println!("(2) DEBUG -- worker terminated");
                 ev_notifier.worker_terminated(child_runtime_name).await;
-                children_specs.push(failed_child_spec);
+                spec_children.push(failed_child_spec);
             }
         }
     }
     if termination_errors.is_empty() {
-        (children_specs, Ok(()))
+        (spec_children, Ok(()))
     } else {
-        (children_specs, Err(TerminationError(termination_errors)))
+        (spec_children, Err(TerminationError(termination_errors)))
+    }
+}
+
+/// terminate_supervisor_monitor triggers the termination of children, and sends
+/// a notification to the client API that spawned the supervisor
+async fn terminate_supervisor_monitor(
+    meta: SpecMeta,
+    sup_runtime_name: String,
+    sup_runtime_children: Vec<Worker>,
+) -> (Spec, Result<(), Arc<TerminationError>>) {
+    // initialize event notifier in this function
+    let mut ev_notifier = meta.ev_notifier.clone();
+
+    // sort runtime children using the spec ordering (LeftToRight, RightToLeft)
+    let runtime_children = meta
+        .order
+        .sort(Lifecycle::Termination, sup_runtime_children);
+
+    let (children_spec, err) = terminate_children(&mut ev_notifier, runtime_children).await;
+
+    // build back Spec to return it to caller
+    let spec = Spec {
+        meta,
+        children: children_spec,
+    };
+
+    match err {
+        Ok(_) => {
```

Comment:
I would be explicit here on the type contained in the `Ok` and just write:
```suggestion
        Ok(()) => {
```

Reply:
I'm ok with this, I'm curious why?